### PR TITLE
Update algorithm table and README

### PR DESCRIPTION
**Pull request by brianhhu**
Minor updates to algorithm table in docs and README to help new users better understand supported saliency algorithms.

### CODE REVIEW

**Comment by brianhhu** [comment]
I was thinking a very similar thing today when editing the table, I will try and update that. Thanks!

**Comment by Purg** [comment]
I *knew* there was a way to use inter-documentation references instead of hard-linking to the readthedocs website. This can be a "problem" in situations like when building your own offline documentation: built docs would then still make requests to the internet. I'm adding a commit that uses cross-referencing to refer to our local docs instead.

**Review by Purg — APPROVED** [review]
Received an LGTM from @brianhhu in an offsite chat.

File: docs/introduction.rst [hunk 1]
```diff
@@ -318,6 +318,7 @@ an algorithm, which is concerned only with the inputs and outputs.
 Additionally, the algorithm can be flexibly determined by the user; that is, the user is free to choose and configure
 the algorithm as needed for the problem domain.
 
+:ref:`custom text <Class: SlidingWindow>`
```
Comment (brianhhu) [inline docs/introduction.rst:321]: @Purg Realized this snuck in on the PR (seems like a leftover artifact from early debugging tests). Let's try to remove this when we update the saliency algorithms table with the new DRISEStack implementation as part of Elim's PR.

**Pull request merged**